### DISCUSSION TITLE
Transforming a camera.toworld with the initial value.

### DISCUSSION THREAD

**Discussion by Whisper-MMM** [mitsuba-renderer/mitsuba3#1002] [Q&A] [answered]
Hi. I'm facing a problem in rendering from multiple view points.
Here is my initial value of the camera.

``` 
PerspectiveCamera.to_world:
 [[[0.00560664, -0.488405, -0.872599, 23.1701],
  [0, 0.872613, -0.488413, 15.7142],
  [0.999984, 0.00274286, 0.00490048, -0.172476],
  [0, 0, 0, 1]]]
```
And I would like to get the parameters and rotate them, then return it with a new value in mi.load_dict.

```
origin = params['PerspectiveCamera.to_world']
rotation = origin.rotate([0.0, 1.0, 0.0], theta)
```
But it seems that I can get no changes from the function.

**Comment by rtabbara** [comment]
Hi @Whisper-MMM ,

I'm not 100% sure if this is what you're after, but from what I can piece together you want something like:

```python
scene = mi.load_dict(...)

params = mi.traverse(scene)

origin = params['PerspectiveCamera.to_world']
rotation = origin.rotate([0.0, 1.0, 0.0], theta)

# Then you want to update the scene?
params['PerspectiveCamera.to_world'] = rotation
params.update()
```

Regardless, there's a really good tutorial [Rendering from multiple view points](https://mitsuba.readthedocs.io/en/latest/src/rendering/multi_view_rendering.html) that shows you how to initialise your multiple sensors independent of the scene which might a better approach anyway.

**Reply by Whisper-MMM** [reply]
Thanks for your kind reply and the excellent work. I'm following the tutorial Rendering from multiple view points. 
And I want to change the coordinates directly using rotate function.

```
def load_sensor(theta):
    origin = params['PerspectiveCamera.to_world']
    rotation = origin.rotate([0.0, 1.0, 0.0], theta)
    print(origin)
    print(rotation)
    return mi.load_dict({
        'type': 'perspective',
        'fov': 32,
        'to_world': rotation,
        'sampler': {
            'type': 'independent',
            'sample_count': 16
        },
        'film': {
            'type': 'hdrfilm',
            'width': 512,
            'height': 512,
            'rfilter': {
                'type': 'tent',
            },
            'pixel_format': 'rgb',
        },
    })
```

```
sensor_count = 6

thetas = [30.0 * i  for i in range(sensor_count)]

sensors = [load_sensor(theta) for theta in thetas]
```

But the result looks like a new matrix is created directly instead of rotating the original matrix.

```
origin:
[[[0.00560664, -0.488405, -0.872599, 23.1701],
  [0, 0.872613, -0.488413, 15.7142],
  [0.999984, 0.00274286, 0.00490048, -0.172476],
  [0, 0, 0, 1]]]
rotation:
[[[1, 0, 0, 0],
  [0, 1, 0, 0],
  [0, 0, 1, 0],
  [0, 0, 0, 1]]]
```

**Reply by rtabbara** [reply, marked as answer]
Oh, I think I know what's going on. Please try

```
rotation = mi.ScalarTransform4f.rotate([0.0, 1.0, 0.0], theta) @ origin
```
`rotate` here is a static method which is why it's returning just the transformation and not transformation applied to `origin`

**Reply by Whisper-MMM** [reply]
I tried this and worked. Thanks a lot!!
